packages/block-editor/src/components/block-switcher/test/index.js

```diff
@@ -211,7 +220,18 @@ describe( 'BlockSwitcherDropdownMenu', () => {
 				} ),
 				'[ArrowDown]'
 			);
-			await act( () => Promise.resolve() );
+
+			const menu = screen.getByRole( 'menu', {
+				name: 'Block Name',
+			} );
+
+			expect( menu ).not.toBeVisible();
```

Comment:
This `.not.toBeVisible()` check points at another thing a reliable test should watch for. The `menu` is invisible not because the popover is not yet positioned, but because its appearance is animated with `framer-motion`. At the very beginning the popover has `opacity: 0`.

The animation and positioning are completely independent from each other.

Reply:
Exactly.